src/MongoDB.Driver.Core/Core/Clusters/ClusterDescription.cs

```diff
@@ -165,6 +180,24 @@ public override int GetHashCode()
                 .GetHashCode();
         }
 
+        /// <summary>
+        /// Selects the servers that determine whether sessions are supported.
+        /// </summary>
+        /// <param name="servers">The servers.</param>
+        /// <returns>The selected servers.</returns>
+        public IEnumerable<ServerDescription> SelectServersThatDetermineWhetherSessionsAreSupported(IEnumerable<ServerDescription> servers)
```

Comment:
This is the central one and only place that determines which servers in a ClusterDescription determine whether sessions are supported.

Reply:
That's quite a mouthful!